runtime/src/main/java/com/flipkart/flux/representation/ClientElbPersistenceService.java

```diff
@@ -0,0 +1,159 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.flipkart.flux.representation;
+
+import com.flipkart.flux.api.ClientElbDefinition;
+import com.flipkart.flux.clientelb.dao.iface.ClientElbDAO;
+import com.flipkart.flux.domain.ClientElb;
+import com.flipkart.flux.persistence.SelectDataSource;
+import com.flipkart.flux.persistence.Storage;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.transaction.Transactional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * <code>ClientElbPersistenceService</code> class does persistence related operations in memory
+ * before storing, updating, reading or deleting objects in DB. It is also responsible for maintaining
+ * an in-memory cache(Google Guava: https://github.com/google/guava/wiki) of clientElb entries for faster query results.
+ *
+ * @author akif.khan
+ */
+@Singleton
+public class ClientElbPersistenceService {
+
+    private ClientElbDAO clientElbDAO;
+
+    private LoadingCache<String, ClientElb> clientElbCache;
+
+    private Integer MAX_CACHE_SIZE;
+
+    private static final Logger logger = LoggerFactory.getLogger(ClientElbPersistenceService.class);
+
+    @Inject
+    public ClientElbPersistenceService(ClientElbDAO clientElbDAO,
+                                       @Named("elbCache.maxSize") Integer MAX_CACHE_SIZE) {
+        this.clientElbDAO = clientElbDAO;
+        this.MAX_CACHE_SIZE = MAX_CACHE_SIZE;
+        this.clientElbCache = CacheBuilder.newBuilder().maximumSize(MAX_CACHE_SIZE)
+                .expireAfterAccess(1, TimeUnit.DAYS)
+                .build(new CacheLoader<String, ClientElb>() {
+                    @Override
+                    public ClientElb load(String clientElbId) throws Exception {
+                        //Call ClientElbDAO layer, if entry not found in cache
+                        return clientElbDAO.findById(clientElbId);
+                    }
+                });
+    }
+
+    public long getClientElbCacheSize() {
+        return this.clientElbCache.size();
+    }
+
+    public boolean clientElbCacheContainsKey(String key) {
+        return this.clientElbCache.asMap().containsKey(key);
+    }
+
+    public String getClientElbCacheUrl(String id) {
+        String clientElbUrl = null;
+        try {
+            clientElbUrl = this.clientElbCache.get(id).getElbUrl();
+        } catch (ExecutionException e) {
+            logger.error("ClientElbCache entry not found in both DB and cache {} {}", e.getMessage(), e.getStackTrace());
```

Comment:
where is the check to db made ?

Reply:
In case of cache miss, LoadingCache makes a call to @override method "load" where we make a DB call to fetch results. Added check for returned "null" value.